src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchResultTableRenderer.cs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal class PackageSearchResultTableRenderer : IPackageSearchResultRenderer
+    {
+        private string _searchTerm;
+        private ILoggerWithColor _loggerWithColor;
+        private const int LineSeparatorLength = 40;
+        private static readonly string SourceSeparator = new('*', LineSeparatorLength);
+        private PackageSearchVerbosity _verbosity;
+        private bool _exactMatch;
+
+        private readonly string[] _minimalVerbosityTableHeader = { "Package ID", "Latest Version" };
+        private readonly string[] _minimalVerbosityTableHeaderForExactMatch = { "Package ID", "Version" };
+        private readonly int[] _minimalColumnsToHighlight = { 0 };
+
+        private readonly string[] _normalVerbosityTableHeader = { "Package ID", "Latest Version", "Owners", "Total Downloads" };
+        private readonly string[] _normalVerbosityTableHeaderForExactMatch = { "Package ID", "Version", "Owners", "Total Downloads" };
+        private readonly int[] _normalColumnsToHighlight = { 0, 2 };
+
+        private readonly string[] _detailedVerbosityTableHeader = { "Package ID", "Latest Version", "Owners", "Total Downloads", "Vulnerable", "Deprecation", "Project URL", "Description" };
+        private readonly string[] _detailedVerbosityTableHeaderForExactMatch = { "Package ID", "Version", "Owners", "Total Downloads", "Vulnerable", "Deprecation", "Project URL", "Description" };
+        private readonly int[] _detailedColumnsToHighlight = { 0, 2, 6, 7 };
+
+        public PackageSearchResultTableRenderer(string searchTerm, ILoggerWithColor loggerWithColor, PackageSearchVerbosity verbosity, bool exactMatch)
+        {
+            _searchTerm = searchTerm;
+            _loggerWithColor = loggerWithColor;
+            _verbosity = verbosity;
+            _exactMatch = exactMatch;
+        }
+
+        public void Add(PackageSource source, IEnumerable<IPackageSearchMetadata> completedSearch)
+        {
+            _loggerWithColor.LogMinimal(SourceSeparator);
+            _loggerWithColor.LogMinimal($"Source: {source.Name} ({source.SourceUri})");
+
+            Table table;
+
+            if (_verbosity == PackageSearchVerbosity.Minimal)
+            {
+                if (_exactMatch)
+                {
+                    table = new Table(_minimalColumnsToHighlight, _minimalVerbosityTableHeaderForExactMatch);
```

Comment:
(Can be addressed afterwards) Consider using the ternary operator ? instead of if statements for assignments. Example:  _exactMatch ? minimalForExactMatch : minimalTableHeader